tsl/src/bgw_policy/compression_api.c

```diff
@@ -118,6 +118,12 @@ policy_compression_add(PG_FUNCTION_ARGS)
 
 	/* check if this is a table with compression enabled */
 	hypertable = ts_hypertable_cache_get_cache_and_entry(ht_oid, CACHE_FLAG_NONE, &hcache);
+
+	if (hypertable_is_distributed(hypertable))
+		ereport(ERROR,
+				(errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+				 errmsg("cannot use compression policy with distributed hypertable")));
```

Comment:
I suggest to add:
```c
errdetail("This limitation might be removed in the future.")
```
as in https://github.com/timescale/timescaledb/pull/2309/files